Task11/src/test/java/com/company/BicycleTest.java

```diff
@@ -1,42 +1,61 @@
 package com.company;
 
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
 
 import java.util.ArrayList;
 
-import static org.junit.Assert.*;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
 
 public class BicycleTest {
     private Bicycle bicycle;
     ArrayList<Point> list;
 
-    @Before
+    @BeforeTest
     public void setUp() throws Exception {
         bicycle = new Bicycle();
         list = new ArrayList<Point>();
         list.add(new Point(0, 0));
     }
-     @After
+
+    @DataProvider (name = "movePoints")
+    public Object[][] moveByCoordinates() {
+        return new Object[][] {
+                {1.4142135623730951, 1, 1},
+                {3.414213562373095, 1, -1},
+                {5.414213562373095, -1, -1},
+                {6.82842712474619, 0, 0}
+        };
+    }
+
+    @Test(dataProvider = "movePoints")
+    public  void testCountDistance(double result, double x, double y) throws Exception {
```

Comment:
dataProvider non-correct, the full route has to be used at one time, points not to be added to list one by one. It seems that the result i simulated